AK/FlyString.cpp

```diff
@@ -96,10 +96,7 @@ bool FlyString::operator==(FlyString const& other) const
 
 bool FlyString::operator==(String const& other) const
 {
-    if (m_data == other)
-        return true;
-
-    return bytes_as_string_view() == other.bytes_as_string_view();
+    return m_data == other;
```

Comment:
Pretty sure this is not correct. And I don't like that this commit is separate from the place where you broke string comparison performance; and it also doesn't fix the other performance issue with the reverse comparison.